extensions/amp-analytics/0.1/scroll-manager.js

```diff
@@ -76,16 +91,28 @@ export class ScrollManager {
   addScrollHandler(handler) {
     // Trigger an event to fire events that might have already happened.
     const size = this.viewport_.getSize();
-    /** {./scroll-manager.ScrollEventDef} */
-    const scrollEvent = {
-      top: this.viewport_.getScrollTop(),
-      left: this.viewport_.getScrollLeft(),
-      width: size.width,
-      height: size.height,
-      scrollWidth: this.viewport_.getScrollWidth(),
-      scrollHeight: this.viewport_.getScrollHeight(),
-    };
-    handler(scrollEvent);
+
+    this.measureRootElement_(true /** initial */).then(layoutRect => {
+      const {
+        top,
+        left,
+        width,
+        height,
+      } = (this.initialRootLayoutRect_ = layoutRect);
+
+      /** {./scroll-manager.ScrollEventDef} */
+      const scrollEvent = {
+        top: this.viewport_.getScrollTop() - top,
```

Comment:
- For clarity/ naming, let's move assignments to the same level. Do the same where you follow this pattern below.

- You're re-assigning `initial` here.

```js
this.getOrMeasureInitialRootRect_().then(rect => {
  const {width, height} = this.viewport_.getSize();
  const {
    width: scrollWidth,
    height: scrollHeight,
  } = rect;
  const scrollEvent = {
    top: this.viewport_.getScrollTop() - rect.top,
    left: this.viewport_.getScrollLeft() - rect.left,
    width,
    height,
    scrollWidth,
    scrollHeight,
    initialScrollLength: {scrollWidth, scrollHeight},
  };
  // ...
});
```